addons/mail/static/src/models/message_tracking_value/message_tracking_value.js

```
@@ -0,0 +1,193 @@
+        }
+
+        /**
+         * @private
+         * @param {string} fieldType
+         * @param {string|boolean|float|number|integer|Array|Object} value
+         * @returns {string}
+         */
+        static _formatMessageTrackingValue(fieldType, value) {
+            /**
+             * Maps tracked field type to a JS formatter. Tracking values are
+             * not always stored in the same field type as their origin type.
+             * Field types that are not listed here are not supported by
+             * tracking in Python. Also see `create_tracking_values` in Python.
+             */
+            switch (fieldType) {
+                case 'boolean':
+                    return format.boolean(value, undefined, { forceString: true });
+                /**
+                 * many2one formatter exists but is expecting id/name_get or data
+                 * object but only the target record name is known in this context.
+                 *
+                 * Selection formatter exists but requires knowing all
+                 * possibilities and they are not given in this context.
+                 */
+                case 'char':
+                case 'many2one':
+                case 'selection':
+                    return format.char(value);
+                case 'date':
+                    if (value) {
+                        value = moment.utc(value);
+                    }
+                    return format.date(value);
+                case 'datetime':
+                    if (value) {
+                        value = moment.utc(value);
+                    }
+                    return format.datetime(value);
+                case 'float':
+                    return format.float(value);
+                case 'integer':
+                    return format.integer(value);
+                case 'text':
+                    return format.text(value);
+                default : 
+                    return undefined;
```

I think this should be an exception, not a silent error.